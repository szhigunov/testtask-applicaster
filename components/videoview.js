import React from 'react';
import {View, StyleSheet} from 'react-native';
import Video from 'react-native-video';

export const VideoView = ({onLoad, onError, uri}) => {
  return (
    <View style={styles.videoView}>
      <Video
        style={styles.video}
        source={{
          uri,
        }}
        rate={1.0}
        volume={0}
        muted={false}
        paused={false}
        resizeMode="contain"
        repeat={false}
        controls={true}
        playInBackground={false}
        playWhenInactive={false}
        progressUpdateInterval={250.0}
        onLoad={() => {
          onLoad(true);
        }}
        onError={err => {
          onError(false);
          console.warn(err);
        }}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  videoView: {
    flex: 1,
    backgroundColor: '#000',
    alignItems: 'center',
    justifyContent: 'center',
  },
  video: {
    width: '100%',
    flex: 1,
  },
});
