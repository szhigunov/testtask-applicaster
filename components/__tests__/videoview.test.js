/**
 * @format
 */

import 'react-native';
import React from 'react';
import {VideoView} from '../videoview';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('VideoView renders correctly', () => {
  let root = renderer.create(<VideoView uri="someurl" />);
  expect(root.toJSON()).toMatchSnapshot();
});
