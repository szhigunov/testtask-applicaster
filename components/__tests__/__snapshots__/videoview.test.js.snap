// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoView renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Video
    controls={true}
    muted={false}
    onError={[Function]}
    onLoad={[Function]}
    paused={false}
    playInBackground={false}
    playWhenInactive={false}
    progressUpdateInterval={250}
    rate={1}
    repeat={false}
    resizeMode="contain"
    source={
      Object {
        "uri": "someurl",
      }
    }
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
    volume={0}
  />
</View>
`;
